cns/service.go

```diff
@@ -25,15 +25,23 @@ import (
 
 const (
 	// Default CNS server URL.
-	defaultAPIServerURL = "tcp://localhost:10090"
-	genericData         = "com.microsoft.azure.network.generic"
+	defaultAPIServerURL  = "tcp://localhost:10090"
+	defaultAPIServerPort = "10090"
+	genericData          = "com.microsoft.azure.network.generic"
+	LocalListener        = "localListener"
+	NodeListener         = "nodeListener"
 )
 
+type cnsListener struct {
+	Listener     acn.Listener
+	ListenerType string
+}
+
 // Service defines Container Networking Service.
 type Service struct {
 	*common.Service
 	EndpointType string
-	Listener     *acn.Listener
+	Listeners    []*cnsListener
```

Comment:
why not `Listeners    []cnsListener`?

Reply:
because when we construct a newListener:
acn.NewListener(nodeURL)
This returns the *acn.Listener